include/geos/simplify/TaggedLineString.h

```diff
@@ -105,7 +105,7 @@ class GEOS_DLL TaggedLineString {
 
     void addToResult(std::unique_ptr<TaggedLineSegment> seg);
 
-    void removeRingEndpoint();
+    TaggedLineSegment* removeRingEndpoint();
```

Comment:
```suggestion
    const TaggedLineSegment* removeRingEndpoint();
```

Reply:
returning a `const` pointer here makes it clear that the caller does not own the returned object.